ngraph/core/src/pass/visualize_tree.cpp

```diff
@@ -349,19 +350,74 @@ static std::string pretty_partial_shape(const PartialShape& shape)
 }
 
 template <typename T>
-static std::string pretty_value(const vector<T>& values, size_t max_elements)
+static std::string pretty_min_max_denormal_value(const vector<T>& values)
 {
     std::stringstream ss;
+
+    T min_value = values[0];
+    T max_value = values[0];
+    size_t denormal_counts = 0ul;
+    std::stringstream denormal_ss;
     for (size_t i = 0; i < values.size(); ++i)
     {
-        if (i != 0 && i % 8 == 0)
+        const auto& value = values[i];
+        if (min_value > value)
         {
-            ss << std::endl;
+            min_value = value;
         }
+        if (max_value < value)
+        {
+            max_value = value;
+        }
+
+        const auto abs_value = std::abs(static_cast<double>(value));
+        if (((abs_value > 0.) && (abs_value < 1.e-32)) || (abs_value > 1.e+32))
+        {
+            if (denormal_counts < 3)
+            {
+                denormal_ss << (denormal_counts > 0 ? ", " : "") << i << ": " << value;
+            }
+            else if (denormal_counts == 3)
+            {
+                denormal_ss << "...";
+            }
+            denormal_counts++;
+        }
+    }
+
+    ss << "min: " << min_value << ", max: " << max_value;
+    if (denormal_counts != 0)
+    {
+        ss << ", denormals: " << denormal_counts << " [" << denormal_ss.str() << "]";
+    }
 
-        if (i >= max_elements)
+    return ss.str();
+}
+
+template <typename T>
+static std::string pretty_value(const vector<T>& values, size_t max_elements)
+{
+    const std::string additional_ss = getenv_bool("NGRAPH_VISUALIZE_TREE_MIN_MAX_DENORMAL")
+                                          ? pretty_min_max_denormal_value(values)
+                                          : "";
+
+    std::stringstream ss;
+    for (size_t i = 0; i < values.size(); ++i)
+    {
+        if (i < max_elements)
+        {
+            if (i != 0 && i % 8 == 0)
+            {
+                ss << std::endl;
+            }
+        }
+        else
         {
             ss << "...";
+            if (!additional_ss.empty())
+            {
+                ss << std::endl << "(" << additional_ss << ")";
```

Comment:
It looks very strange. Please move `if (!additional_ss.empty())` after this cycle.
Also I expect that the number of denormal values will be equal to the number of values printed in prety_value function.

Reply:
> It looks very strange. Please move if (!additional_ss.empty()) after this cycle.

Done: fixed, thanks!
> Also I expect that the number of denormal values will be equal to the number of values printed in prety_value function.

We display amount of denormal values and 3 values only. It was enough for me for a model enabling.

Reply:
From my point of view we don't need to have another code that traverse vector and convert denormal values. I suggest to have a simple function that works with single value but not with vector.
Also your point that amount of values 3 is enough is a bit confusing. I suggest to limit the number of denormal values in the same way as for other values.

Reply:
>From my point of view we don't need to have another code that traverse vector and convert denormal values. I suggest to have a simple function that works with single value but not with vector.

First, we are speaking about visualization, correct? We need to visualize if denormal values exist or not, there is no conversion here. Second, denormal visualization is optional and doesn't affect anything by default.

>Also your point that amount of values 3 is enough is a bit confusing. I suggest to limit the number of denormal values in the same way as for other values.

`Constant` operation is displayed as a box. 3 values and their indexes are enough to fill available space after min/max and the number of denormal values. We need to display some denormal values and their indexes to track these values operation by operation in the dump files.